src/modules/pair/services/pair.compute.service.ts

```diff
@@ -543,4 +555,102 @@ export class PairComputeService implements IPairComputeService {
                 .multipliedBy(firstTokenPriceUSD);
         }
     }
+
+    @ErrorLoggerAsync({
+        logArgs: true,
+    })
+    @GetOrSetCache({
+        baseKey: 'pair',
+        remoteTtl: CacheTtlInfo.ContractState.remoteTtl,
+        localTtl: CacheTtlInfo.ContractState.localTtl,
+    })
+    async hasFarms(pairAddress: string): Promise<boolean> {
+        return await this.computeHasFarms(pairAddress);
+    }
+
+    async computeHasFarms(pairAddress: string): Promise<boolean> {
+        const addresses: string[] = farmsAddresses(['v2']);
+        const lpTokenID = await this.pairAbi.lpTokenID(pairAddress);
+
+        const farmingTokenIDs = await Promise.all(
+            addresses.map((address) =>
+                this.farmAbi.useAbi(address).farmingTokenID(address),
+            ),
+        );
+
+        return farmingTokenIDs.includes(lpTokenID);
+    }
+
+    @ErrorLoggerAsync({
+        logArgs: true,
+    })
+    @GetOrSetCache({
+        baseKey: 'pair',
+        remoteTtl: CacheTtlInfo.ContractState.remoteTtl,
+        localTtl: CacheTtlInfo.ContractState.localTtl,
+    })
+    async hasDualFarms(pairAddress: string): Promise<boolean> {
+        return await this.computeHasDualFarms(pairAddress);
+    }
+
+    async computeHasDualFarms(pairAddress: string): Promise<boolean> {
+        const stakingAddresses =
+            await this.remoteConfigGetterService.getStakingProxyAddresses();
+
+        const pairAddresses = await Promise.all(
+            stakingAddresses.map((address) =>
+                this.stakingProxyAbiService.pairAddress(address),
+            ),
+        );
+
+        return pairAddresses.includes(pairAddress);
+    }
+
+    @ErrorLoggerAsync({
+        logArgs: true,
+    })
+    @GetOrSetCache({
+        baseKey: 'pair',
+        remoteTtl: CacheTtlInfo.ContractState.remoteTtl,
+        localTtl: CacheTtlInfo.ContractState.localTtl,
+    })
+    async tradesCount(pairAddress: string): Promise<number> {
+        return await this.computeTradesCount(pairAddress);
+    }
+
+    async computeTradesCount(pairAddress: string): Promise<number> {
+        const elasticQueryAdapter: ElasticQuery = new ElasticQuery();
+
+        elasticQueryAdapter.condition.must = [
```

Comment:
are we interested in successful swaps or anytime a swap transaction reaches the chain? since I can (as a pair owner) send a lot of failed transactions (100k gas or something) that will boost my pairs in the analytics.

Reply:
added constraint on `status`